### PR TITLE
🐛 don't customize non-manager deployments

### PR DESCRIPTION
**Pull request by Fedosin**
**What this PR does / why we need it**:
    
To prevent customization of non-manager deployments, we introduce a check that first verifies if there are several deployments. If there are several deployments available, we customize only those with names like "ca*-controller-manager" and skip the others.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: internal/controller/component_customizer.go [hunk 1]
```diff
@@ -74,6 +74,18 @@ func customizeObjectsFn(provider operatorv1.GenericProvider) func(objs []unstruc
 			}
 
 			if o.GetKind() == deploymentKind {
+				// We need to skip the deployment customization if there are several deployments available
```
[the remark this answers is not in the document]
Comment (Danil-Grigorev) [reply internal/controller/component_customizer.go:77]: Ah, I see, we are skipping everything which does not follow ca*-controller-manager and has multiple replicas. We’d need to watch out for non-regular providers, if they follow the naming pattern.

/lgtm

**Comment by Fedosin** [comment]
/hold

going to add more tests

File: internal/controller/component_customizer.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Fedosin) [reply internal/controller/component_customizer.go:77]: I simplified that check, now it looks like `if isMultipleDeployments && !isProviderManagerDeploymentName(o.GetName())` and isProviderManagerDeploymentName is `strings.HasPrefix(name, "ca") && strings.HasSuffix(name, "-controller-manager")`

**Review by Danil-Grigorev — COMMENTED** [review]
/lgtm

**Pull request merged**